src/core/EdFi.DataManagementService.Core/Middleware/ValidateRepeatedPropertiesMiddleware.cs

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using EdFi.DataManagementService.Core.Response;
+using Microsoft.Extensions.Logging;
+using static EdFi.DataManagementService.Core.Response.FailureResponse;
+
+namespace EdFi.DataManagementService.Core.Middleware;
+
+internal class ValidateRepeatedPropertiesMiddleware(ILogger logger) : IPipelineStep
+{
+    private static readonly JsonSerializerOptions _serializerOptions =
+        new() { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping };
+
+    public async Task Execute(PipelineContext context, Func<Task> next)
+    {
+        logger.LogDebug(
+            "Entering ValidateRepeatedPropertiesMiddleware - {TraceId}",
+            context.FrontendRequest.TraceId
+        );
+        if (context.FrontendRequest.Body != null)
+        {
+            try
+            {
+                var jsonDocument = JsonDocument.Parse(context.FrontendRequest.Body);
+                var validationErrors = new Dictionary<string, List<string>>();
+
+                CheckForDuplicateProperties(jsonDocument.RootElement, "", validationErrors);
```

Comment:
In ParseBodyMiddleware the string body is taken and converted to `JsonNode JsonNode? body = JsonNode.Parse(context.FrontendRequest.Body);` and although there is an Exception it is not caught by the try catch.
![image](https://github.com/user-attachments/assets/379b8986-fffe-4380-88ee-1b4343c50cd6)